plugin/pkg/auth/authenticator/token/webhook/webhook.go

```diff
@@ -42,55 +41,87 @@ const retryBackoff = 500 * time.Millisecond
-func New(kubeConfigFile string, ttl time.Duration) (*WebhookTokenAuthenticator, error) {
-	return newWithBackoff(kubeConfigFile, ttl, retryBackoff)
+// NewFromInterface creates a webhook authenticator using the given tokenReview client
+func NewFromInterface(tokenReview client.TokenReviewInterface, ttl time.Duration) (*WebhookTokenAuthenticator, error) {
+	return newWithBackoff(tokenReview, ttl, retryBackoff)
 }
 
-// newWithBackoff allows tests to skip the sleep.
-func newWithBackoff(kubeConfigFile string, ttl, initialBackoff time.Duration) (*WebhookTokenAuthenticator, error) {
-	gw, err := webhook.NewGenericWebhook(kubeConfigFile, groupVersions, initialBackoff)
+// New creates a new WebhookTokenAuthenticator from the provided kubeconfig file.
+func New(kubeConfigFile string, ttl time.Duration) (*WebhookTokenAuthenticator, error) {
+	tokenReview, err := tokenReviewClientFromKubeconfig(kubeConfigFile)
 	if err != nil {
 		return nil, err
 	}
-	return &WebhookTokenAuthenticator{gw, cache.NewLRUExpireCache(1024), ttl}, nil
+	return newWithBackoff(tokenReview, ttl, retryBackoff)
+}
+
+// newWithBackoff allows tests to skip the sleep.
+func newWithBackoff(tokenReview client.TokenReviewInterface, ttl, initialBackoff time.Duration) (*WebhookTokenAuthenticator, error) {
+	return &WebhookTokenAuthenticator{tokenReview, cache.NewLRUExpireCache(1024), ttl, initialBackoff}, nil
 }
 
 // AuthenticateToken implements the authenticator.Token interface.
 func (w *WebhookTokenAuthenticator) AuthenticateToken(token string) (user.Info, bool, error) {
-	r := &v1beta1.TokenReview{
-		Spec: v1beta1.TokenReviewSpec{Token: token},
+	r := &authentication.TokenReview{
+		Spec: authentication.TokenReviewSpec{Token: token},
 	}
 	if entry, ok := w.responseCache.Get(r.Spec); ok {
-		r.Status = entry.(v1beta1.TokenReviewStatus)
+		r.Status = entry.(authentication.TokenReviewStatus)
 	} else {
-		result := w.WithExponentialBackoff(func() restclient.Result {
-			return w.RestClient.Post().Body(r).Do()
+		var (
+			result *authentication.TokenReview
+			err    error
+		)
+		webhook.WithExponentialBackoff(w.initialBackoff, func() error {
+			result, err = w.tokenReview.Create(r)
+			return err
 		})
-		if err := result.Error(); err != nil {
-			return nil, false, err
-		}
-		var statusCode int
-		if result.StatusCode(&statusCode); statusCode < 200 || statusCode >= 300 {
-			return nil, false, fmt.Errorf("Error contacting webhook: %d", statusCode)
-		}
-		spec := r.Spec
-		if err := result.Into(r); err != nil {
+		if err != nil {
 			return nil, false, err
 		}
-		w.responseCache.Add(spec, r.Status, w.ttl)
+		r.Status = result.Status
+		w.responseCache.Add(r.Spec, result.Status, w.ttl)
 	}
 	if !r.Status.Authenticated {
 		return nil, false, nil
 	}
+
+	var extra map[string][]string
+	if r.Status.User.Extra != nil {
+		extra = map[string][]string{}
+		for k, v := range r.Status.User.Extra {
+			extra[k] = ([]string)(v)
```

Comment:
That form `(T)(var)` is evaluated at compile time, not runtime. It wouldn't even build if `v` wasn't always castable to `[]string`.